doc/neps/nep-0049.rst

```diff
@@ -134,6 +134,16 @@ NumPy C-API functions
             void (*free) (void *ctx, void *ptr, size_t size);
         } PyDataMemAllocator;
 
+    The use of a ``size`` parameter in ``free`` differentiates this struct from
+    the :c:type:`PyMemAllocatorEx` struct in Python. This call signature is
+    used internally in NumPy currently, and also in other places for instance
+    C++11 (maybe older): https://en.cppreference.com/w/cpp/memory/allocator/deallocate
+    C++17: https://en.cppreference.com/w/cpp/memory/polymorphic_allocator/deallocate
+    Rust (allocator_api): https://doc.rust-lang.org/std/alloc/trait.Allocator.html#tymethod.deallocate
```

Comment:
```suggestion
    `C++98 <https://en.cppreference.com/w/cpp/memory/allocator/deallocate>`
    `C++11 <https://en.cppreference.com/w/cpp/memory/allocator_traits/deallocate>`
    `Rust (allocator_api) <https://doc.rust-lang.org/std/alloc/trait.Allocator.html#tymethod.deallocate>`
```
`polymorphic_allocator` is just an implementation of `allocator_traits`, so probably isn't interesting here.